x/claim/keeper/hooks.go

```diff
@@ -0,0 +1,89 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	epochstypes "github.com/Stride-Labs/stride/x/epochs/types"
+	stakingibctypes "github.com/Stride-Labs/stride/x/stakeibc/types"
+
+	"github.com/Stride-Labs/stride/x/claim/types"
+)
+
+func (k Keeper) AfterDelegationModified(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress) {
+	identifiers := k.GetAirdropIdentifiersForUser(ctx, delAddr)
+	for _, identifier := range identifiers {
+		k.ClaimCoinsForAction(ctx, delAddr, types.ActionDelegateStake, identifier)
+	}
+}
+
+func (k Keeper) AfterLiquidStake(ctx sdk.Context, addr sdk.AccAddress) {
+	identifiers := k.GetAirdropIdentifiersForUser(ctx, addr)
+	for _, identifier := range identifiers {
+		k.ClaimCoinsForAction(ctx, addr, types.ActionLiquidStake, identifier)
+	}
+}
+
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochInfo epochstypes.EpochInfo) {
+}
+
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochInfo epochstypes.EpochInfo) {
+	k.ClearClaimedStatus(ctx, epochInfo.Identifier)
+}
+
+// ________________________________________________________________________________________
+
+// Hooks wrapper struct for claim keeper
+type Hooks struct {
+	k Keeper
+}
+
+var _ stakingtypes.StakingHooks = Hooks{}
+var _ stakingibctypes.StakeIBCHooks = Hooks{}
+var _ epochstypes.EpochHooks = Hooks{}
+
+// Return the wrapper struct
+func (k Keeper) Hooks() Hooks {
+	return Hooks{k}
+}
+
+// ibcstaking hooks
+func (h Hooks) AfterLiquidStake(ctx sdk.Context, addr sdk.AccAddress) {
+	h.k.AfterLiquidStake(ctx, addr)
+}
+
+// epochs hooks
+func (h Hooks) BeforeEpochStart(ctx sdk.Context, epochInfo epochstypes.EpochInfo) {
+	h.k.BeforeEpochStart(ctx, epochInfo)
+}
+
+func (h Hooks) AfterEpochEnd(ctx sdk.Context, epochInfo epochstypes.EpochInfo) {
+	h.k.AfterEpochEnd(ctx, epochInfo)
+}
+
+// staking hooks
```

Comment:
Yes, we are currently using only `AfterDelegationModified` hook interface method but need to define all staking hook interface methods.
Otherwise, we can't set hooks of claim module to staking module.